src/ADAL.PCL/Internal/Telemetry/OneDriveAggregator.cs

```diff
@@ -0,0 +1,93 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal class OneDriveAggregator : DefaultAggregator
+    {
+        internal IDictionary<string, List<EventsBase>> ObjectsToBeDispatched =
+            new ConcurrentDictionary<string, List<EventsBase>>();
+
+        internal OneDriveAggregator(IDispatcher dispatcher) : base(dispatcher)
+        {
+            Dispatcher = dispatcher;
+        }
+
+        internal override void Flush(string requestID)
+        {
+            Dictionary<string, string> FlatList = new Dictionary<string, string>();
+            Dictionary<string, string> DispatchMap = new Dictionary<string, string>();
+
+            if (ObjectsToBeDispatched == null || (ObjectsToBeDispatched.Count == 0))
+            {
+                return;
+            }
+
+            foreach (KeyValuePair<string, List<EventsBase>> pair in ObjectsToBeDispatched)
+            {
+                if (requestID.Equals(pair.Key))
+                {
+                    foreach (EventsBase Event in pair.Value)
+                    {
+                        Event.ProcessEvent(DispatchMap);
+                    }
+                }
+            }
+
+            ObjectsToBeDispatched.Clear();
```

Comment:
Did you clear the ObjectsToBeDispatched completely here? If you are tracking 2 request, you will dispatch one and drop the other? Or am i reading this wrong?